sphinxcontrib/images.py

```diff
@@ -1,53 +1,52 @@
-# -*- coding: utf-8 -*-
-__version__ = '1.0.1'
-__author__ = 'Tomasz Czyż <tomaszczyz@gmail.com>'
-__license__ = "Apache 2"
+from __future__ import annotations
 
-
-import os
-import sys
+import argparse
 import copy
-import uuid
+import functools
 import hashlib
 import importlib.metadata
-import argparse
-import functools
-
+import os
+import requests
 import sphinx
-from sphinx.util.osutil import copyfile
-
-try:
-    from sphinx.util import logging
-
-    logger = logging.getLogger(__name__)
-except ImportError:
-    logger = None
+import sys
+import uuid
 
-try:
-    from sphinx.util.compat import Directive
-except:
-    from docutils.parsers.rst import Directive
+from collections.abc import Iterable
+from docutils import nodes
+from docutils.parsers.rst import Directive, directives
+from sphinx.util import logging, osutil
+from typing import Any, Literal, Protocol, TYPE_CHECKING
 
 try:
     from sphinx.util.display import status_iterator
 except ImportError:
-    try:
-        from sphinx.util import status_iterator
-    except ImportError:
-        pass
```

Comment:
`app.status_iterator` was removed in Sphinx 1.7

Reply:
Yup, your memory is correct! Removing this support allowed simplifying those lines to just use `status_iterator` as a utility function only rather than mixing between that and a method.